server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -801,11 +803,19 @@ protected Node(
                     List.of(remoteIndexPathUploader),
                     namedWriteableRegistry
                 );
+                remoteStorePinnedTimestampService = new RemoteStorePinnedTimestampService(
```

Comment:
Not completely sure if we want to keep this behind cluster state enabled setting.

Reply:
Agree, let me put it behind remote store settings only as this is remote store feature.